### PR TITLE
New mayhem, add wipe option, and more

### PR DESCRIPTION
**Pull request by mprimi**

### CODE REVIEW

File: run-test.sh [hunk 1]
```diff
@@ -103,6 +104,22 @@ function random_reload() {
...
+# Mayhem function random_hard_kill randomly kills (SIGKILL) one of the servers
+MIN_TIME_BETWEEN_HARD_KILL=1 # Minimum time between kills
+MAX_TIME_BETWEEN_HARD_KILL=10 # Maximum time between kills
+function random_hard_kill() {
+  while true; do
+    RANDOM_POD="nats-$(( ${RANDOM} % ${CLUSTER_SIZE} ))"
+    RANDOM_DELAY="$(( (${RANDOM} % (${MAX_TIME_BETWEEN_HARD_KILL} - ${MIN_TIME_BETWEEN_HARD_KILL})) + ${MIN_TIME_BETWEEN_HARD_KILL} ))"
+    echo "🐵 Killing ${RANDOM_POD} (sigkill)"
+    kubectl exec "pod/${RANDOM_POD}" -c nats quiet -- sh -c 'kill -9 $(cat /var/run/nats/nats.pid)' || echo "Failed to kill ${RANDOM_POD}"
+    # TODO: wait for all pods to be running again using
```
Comment (ReubenMathew) [inline run-test.sh:117]: Would a hard kill occur while a test is running? Or would it wait till the cluster stare stabilizes to run a workload?

File: tests/main.go [hunk 2]
```diff
@@ -14,29 +14,43 @@ var options struct {
...
+	}
+	testsMap[testName] = testFunc
 }
 
 func main() {
 
 	flag.DurationVar(&options.TestDuration, "duration", 60*time.Second, "How long to run")
 	flag.StringVar(&options.ServerURL, "server", nats.DefaultURL, "Server URL")
 	flag.StringVar(&options.TestName, "test", "", "name of test")
+	flag.BoolVar(&options.Wipe, "wipe", false, "Delete all resources before starting test")
```
Comment (ReubenMathew) [inline tests/main.go:34]: This is great! I've just been running a test twice so it can fail the first time and cleanup.

**Review by ReubenMathew — APPROVED** [review]

**Comment by ReubenMathew** [comment]
Just for my curiosity, would this be a squash and merge scenario? Your individual commits look pretty focused on a singular thing. Would a full squash be necessary?

File: run-test.sh (hunk 1 above)
Comment (mprimi) [reply run-test.sh:117]: as you can see, the logic is currently 

```
while true
  kill $RANDOM_NODE
  sleep $RANDOM
```

And that is running while the test is running. 

The TODO is to let the previously killed node come back before trying to kill a different one.

File: tests/main.go (hunk 2 above)
Comment (mprimi) [reply tests/main.go:34]: > would this be a squash and merge scenario?

No, commits are doing different things. 

This was originally 10+ tiny and messy commits that i already squashed into 5 atomic changes.

**Comment by mprimi** [comment]
> would this be a squash and merge scenario?

No, commits are doing different things.

This was originally 10+ tiny and messy commits that i already squashed into 5 atomic changes.

**Pull request merged**